torch_tvm/compiler.cpp

```diff
@@ -382,6 +422,15 @@ void TVMCompiler::run(Stack& stack) {
     cache_[spec].kernel = run_mod.GetFunction("run", false);
     cache_[spec].get_output = run_mod.GetFunction("get_output", false);
     auto get_num_outputs = run_mod.GetFunction("get_num_outputs", false);
+
+    // Set parameter inputs.
+    tvm::Map<std::string, tvm::relay::Constant> params = get_params();
+    for (const auto& param : params) {
+        const auto& param_name = param.first;
+        const auto& param_ndarray_val = param.second->data;
+        cache_[spec].set_input(param_name, param_ndarray_val);
+    }
```

Comment:
Are we setting the input value for params node twice? Assume we have done it before the build, right?

Reply:
Also this function is too large, I think we had better to refactor it.

Reply:
Yes the function is growing. Although I just did a refactor by moving set_input into a separate function. But regarding your question on setting params twice. We need to do this because the first time before build it is just to tell what are the param nodes. I am not sure at this point if it creates only the precompute net or also populates the weight value. I did try not setting it again and it was resulting in wrong value. After the build process, we need to set the values again but only once. This is certainly not efficient and can be improved upon on the tvm side, I suppose.

Reply:
hmm, If that's the case, how can we do the pre-compute pass in the building process. I think we need the concrete value.

Reply:
Well for my case it probably was necessary because there is nothing that you can precompute. Since the weight packing stuff is inside topi op it cannot be precomputed so I still need to provide the relevant data pointer.

Reply:
But either way, I dont think setting it again will mess with precompute if any could be done.

Reply:
I guess there might be some bug/issue for constant node.